department-of-veterans-affairs/caseflow
# Schedule Veteran Page Style Changes

Resolves [CASEFLOW-1488](https://vajira.max.gov/browse/CASEFLOW-1488)

### Description
The goal is to improve the layout of the Schedule Veteran page.

### Acceptance Criteria
- [x] Adjust widths
    - [x] Make the length of all input fields consistent
    - [x] Make the divider lines span the width of the section that contains the input fields (while respecting the existing left / right padding)
    - [x] [Added by Charley] Wrap HelperText so it doesn't extend beyond the dropdown it refers to.
- [x] Adjust Heights
    - [x] Make padding between all input fields consistent (30px seems standard throughout Caseflow)
    - [x] Reduce the line-height of the help text on the timezone and email input fields (1.5em seems standard for body copy  according to Storybook)
- [x] Fix select alignment. This problem actually appears on many of our other dropdowns in a less extreme form.
    - [x] On custom time modal, center selected time in the field

### Testing Plan
1. Reviewing the screenshots attached here and on the Jira Ticket is a good way to see before/after.
2. To test ScheduleVeteran in a development environment
    1. Create a virtual hearing day by adding a hearing day through the UI.
    2. Open the scheduling screen for a veteran, select the hearing day you just added, set the type to 'Virtual'. You should see time slots.
    3. Don't click 'schedule'
    3. Change the hearing day to a video day, set the type to 'Video' time slots will disappear.
    4. Review the layout of both of those screens.
3. There are two other places these changes might effect the layout
    1. Hearing Conversion can be gotten to from details from a docket, url [like this](http://localhost:3000/hearings/12e1d059-d626-4271-afb0-3850c095ca67/details).
    2. HearingTypeConversionModal, the trick is to choose a veteran where the hearing is NOT scheduled yet, then in the same dropdown as schedule veteran select 'Convert to Video'/'Convert to Virtual'. Url is [like this](http://localhost:3000/queue/appeals/b42edc58-987f-460b-991e-339eb564f9e8/tasks/975/change_hearing_request_type_to_virtual). Steps to get to this part of the UI are in this PR: https://github.com/department-of-veterans-affairs/caseflow/pull/15847  

 BEFORE|AFTER
 ---|---
![before_with_slots](https://user-images.githubusercontent.com/6818839/123819407-c0393c00-d8c7-11eb-8a16-d92d7a53352c.png) ![before_no_slots](https://user-images.githubusercontent.com/6818839/123819409-c0d1d280-d8c7-11eb-85d1-73d1b55871b4.png) | ![after_1](https://user-images.githubusercontent.com/6818839/124022822-54380000-d9bb-11eb-93cf-74bea398db4d.png) ![after_2](https://user-images.githubusercontent.com/6818839/124022835-569a5a00-d9bb-11eb-9956-f1266d7f4cbf.png)
